allennlp/modules/token_embedders/pretrained_transformer_mismatched_embedder.py

```diff
@@ -111,15 +119,53 @@ def forward(
         # span_embeddings: (batch_size, num_orig_tokens, max_span_length, embedding_size)
         # span_mask: (batch_size, num_orig_tokens, max_span_length)
         span_embeddings, span_mask = util.batched_span_select(embeddings.contiguous(), offsets)
-        span_mask = span_mask.unsqueeze(-1)
-        span_embeddings *= span_mask  # zero out paddings
 
-        span_embeddings_sum = span_embeddings.sum(2)
-        span_embeddings_len = span_mask.sum(2)
-        # Shape: (batch_size, num_orig_tokens, embedding_size)
-        orig_embeddings = span_embeddings_sum / torch.clamp_min(span_embeddings_len, 1)
+        # int: maximum number of sub-tokens a batch has
+        max_span_length = span_mask.shape[-1]
+
+        # If "sub_token_mode" is set to "first", return the first sub-token embedding
+        if self.sub_token_mode == "first":
+            # Shape: (batch_size, num_orig_tokens, max_span_length)
+            span_mask_bool = torch.zeros(span_mask.shape, dtype=torch.bool)
+
+            # Shape: 1-dimensional tensor
+            max_span_range_indices = torch.arange(max_span_length, dtype=torch.long).view(1, 1, -1)
+
+            # Shape (batch_size, num_orig_tokens, max_span_length)
+            span_mask = max_span_range_indices <= span_mask_bool
+            span_mask = span_mask.unsqueeze(-1)
+
+            # Shape: (batch_size, num_orig_tokens, max_span_length, embedding_size)
+            span_embeddings *= span_mask
+
+            # Sum over embeddings of all sub-tokens of a word where except first sub-token,
+            # rest all sub-token's embeddings is zero padded
+            # Shape: (batch_size, num_orig_tokens, embedding_size)
+            orig_embeddings = span_embeddings.sum(2)
```

Comment:
I don't have the complete overview over this code, but doesn't this boil down to `orig_embeddings = span_embeddings[..., 0]`?

I might be missing a dimension somewhere.

Reply:
Here `span_embeddings` is of shape `(batch_size, num_orig_tokens, max_span_length, embedding_size)` where from the 3rd dimension i.e. `max_span_length`, only first tensor has actual values (first subtoken embedding), rest have zero elements. 

When sum is taken along 3rd dimension i.e. `max_span_length` that will boil down to shape `(batch_size, num_orig_tokens, embedding_size)` and hence one of the dimension will be squeezed automatically by sum function

Reply:
So then `orig_embeddings = span_embeddings[..., 0, :]`, right? Why mess around with `span_mask` and `span_mask_bool` and all that?

Reply:
Oh wait, you already answered that below. I should read ahead :-)